packages/datastore/src/util.ts

```diff
@@ -681,3 +682,71 @@ export const isAWSIPAddress = (val: string): boolean => {
 		val
 	);
 };
+
+export function DeferredPromise(): void {
+	const self = this;
+	this.promise = new Promise((resolve, reject) => {
+		self.resolve = resolve;
+		self.reject = reject;
+	});
+}
+
+export class SubscriptionBuffer {
+	private limitPromise = new DeferredPromise();
+	private timerPromise: Promise<string>;
+	private TIMER_INTERVAL = 2000;
+	private timer: NodeJS.Timer;
+	private raceInFlight = false;
+	private callback = () => {};
+	private errorHandler: (error: string) => void;
+	private defaultErrorHandler = (msg = 'SubscriptionBuffer error'): void => {
+		throw new Error(msg);
+	};
+
+	constructor(options: SubscriptionBufferOptions) {
+		this.callback = options.callback;
+		this.errorHandler = options.errorHandler || this.defaultErrorHandler;
+		this.TIMER_INTERVAL = options.maxInterval;
```

Comment:
Same issue as we were seeing previously with the errorHandler. `options.maxInterval` will always overwrite `TIMER_INTERVAL` here. If we want it to default to 2000, we'd want to do:
```suggestion
		this.timerInterval = options.maxInterval || 2000;
```
Also, no need to make that property all caps, since it's configurable